### ISSUE TITLE
wrong config deserialization resulting in `Error on fetch for [object Object]` when using conditional map

### ISSUE THREAD

**Issue by mikz** [jspm/jspm-cli#1501]
the code is available in https://github.com/mikz/jspm-config-deserialization-error
the code in question is made as one commit https://github.com/mikz/jspm-config-deserialization-error/commit/8baf75c46a3ed743a3881fb020f39a26996908a3

Maybe I'm missing something obvious, But this is almost the same as the example in http://jspm.io/0.17-beta-guide/conditional-loading.html

I was tracking it down and ended in https://github.com/jspm/jspm-cli/blob/63ae833bd68430c328a4539ad85420e33791b1c2/lib/config/config-file.js
For some reason, the properties were deserialized properly, but then serialized into `"[object Object]"`.
By the time this gets to SystemJS it is already wrong.

Bundling the code produces following error:
```
$ jspm bundle code.js --production
Building the bundle tree for code.js...

err  Error on fetch for [object Object] at file:///Users/mikz/Developer/3scale/system/jspmtest/[object Object]
  Loading code.js
  Loading hot-reload/index.js
  Error: ENOENT: no such file or directory, open '/Users/mikz/Developer/3scale/system/jspmtest/[object Object]'
    at Error (native)
```

Because in `SystemJS.packages['hot-reload']` the `map` is already a string "[object Object]".

Btw, this flow would be significantly improved by https://github.com/systemjs/systemjs/issues/946.

**Comment by guybedford** [comment]
I tried to replicate with:

```javascript
  packages: {
    "app": {
      "main": "index.js",
      "map": {
        "./index.js": {
          "production": "./index-production.js"
        }
      }
    }
  }
```

and `jspm bundle app --production`, but wasn't able to. Can you clarify which jspm version you are using here? Are you sure it is beta 6?

**Comment by guybedford** [comment]
Ok I was able to get the replication from the test repo here, and added the fixes in https://github.com/jspm/jspm-cli/commit/77a30b7cbadb866c8bfc9707cbb34d45028f3fc7. Thanks for these cases!

**Comment by mikz** [comment]
@guybedford tried that and can confirm it works! Thank you very much.

**Comment by guybedford** [comment]
Fixed in jspm 0.17.0-beta.7.
